lib/internal/modules/cjs/loader.js

```diff
@@ -1293,5 +1295,12 @@ Module.syncBuiltinESMExports = function syncBuiltinESMExports() {
   }
 };
 
+Module.isBuiltIn = function isBuiltIn(moduleName) {
+  return ArrayPrototypeIncludes(
+    [...Module.builtinModules,...NativeModule.getSchemeOnlyModuleNames()],
```

Comment:
These two can't be merged like this because `NativeModule.getSchemeOnlyModuleNames()` returns unprefixed names and they are _only_ available with a prefix, so I think you will need two separate checks: one for modules that are available with or without a prefix and another for modules that require `moduleName` to include the prefix.

Reply:
hmm, interesting. So, something like the below should do? @mscdex 

```
ArrayPrototypeIncludes(
    Module.builtinModules,
    StringPrototypeReplace(String(moduleName), /^node:/, '')
  ) || ArrayPrototypeIncludes(
    Module.builtinModules,
    NativeModule.getSchemeOnlyModuleNames(),
    StringPrototypeReplace(String(moduleName), /^node:/, '')
  )
  ```

Reply:
The second check has to make sure that `moduleName` has the `'node:'` prefix. There's a variety of ways to implement the actual check against the list returned by `getSchemeOnlyModuleNames()`, but that's the important part.

Reply:
Okies, [updated](https://github.com/nodejs/node/pull/43396/commits/420c8ebe24c8011d83b60b4f2202364c92e0d63c).